utils/virtual_machines.py

```diff
@@ -44,30 +47,35 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
     logger.info("Getting ready to deploy VM/instance %s from template %s on provider %s" %
         (vm_name, template_name, data['name']))
     try:
-        logger.debug("Deploy args: " + str(deploy_args))
-        vm_name = mgmt.deploy_template(template_name, timeout=timeout, **deploy_args)
-        logger.info("Provisioned VM/instance %s" % vm_name)  # instance ID in case of EC2
-    except Exception as e:
-        logger.error('Could not provisioning VM/instance %s (%s)', vm_name, e)
         try:
-            logger.info("VM/Instance status: {}".format(mgmt.vm_status(vm_name)))
-        except Exception as f:
-            logger.error(
-                "Could not retrieve VM/Instance status: {}: {}".format(type(f).__name__, str(f)))
-        logger.info('Attempting cleanup on VM/instance %s', vm_name)
-        try:
-            if mgmt.does_vm_exist(vm_name):
-                # Stop the vm first
-                logger.warning('Destroying VM/instance %s', vm_name)
-                if mgmt.delete_vm(vm_name):
-                    logger.info('VM/instance %s destroyed', vm_name)
-                else:
-                    logger.error('Error destroying VM/instance %s', vm_name)
-        except Exception as f:
-            logger.error(
-                'Could not destroy VM/instance {} ({}: {})'.format(
-                    vm_name, type(f).__name__, str(f)))
-        finally:
-            raise e
-
+            logger.debug("Deploy args: " + str(deploy_args))
+            vm_name = mgmt.deploy_template(template_name, timeout=timeout, **deploy_args)
+            logger.info("Provisioned VM/instance %s" % vm_name)  # instance ID in case of EC2
+        except Exception as e:
+            logger.error('Could not provisioning VM/instance %s (%s)', vm_name, e)
+            try:
+                logger.info("VM/Instance status: {}".format(mgmt.vm_status(vm_name)))
+            except Exception as f:
+                logger.error(
+                    "Could not retrieve VM/Instance status: {}: {}".format(
+                        type(f).__name__, str(f)))
+            logger.info('Attempting cleanup on VM/instance %s', vm_name)
+            try:
+                if mgmt.does_vm_exist(vm_name):
+                    # Stop the vm first
+                    logger.warning('Destroying VM/instance %s', vm_name)
+                    if mgmt.delete_vm(vm_name):
+                        logger.info('VM/instance %s destroyed', vm_name)
+                    else:
+                        logger.error('Error destroying VM/instance %s', vm_name)
+            except Exception as f:
+                logger.error(
+                    'Could not destroy VM/instance {} ({}: {})'.format(
+                        vm_name, type(f).__name__, str(f)))
+            finally:
+                raise e
+    except RHEVRequestError as e:
```

Comment:
Maybe change the name, but I think I'd rather see `allow_skip` to be a list of exceptions that trigger a skip, or maybe a list of callables that return True/False if a skip should happen based on the exception, or maybe both, or something else you devise. :)